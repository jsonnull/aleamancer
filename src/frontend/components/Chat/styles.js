// @flow
import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${props => props.theme.background};
  width: 320px;
  overflow: hidden;
  border-radius: ${props => (props.isPinned ? '0' : '5px')};
  box-shadow: ${props =>
    props.isPinned ? '' : '0 3px 5px 0 rgba(0, 0, 0, 0.2)'};

  height: ${props => (props.isPinned ? '100%' : 'auto')};
  position: ${props => (props.isPinned ? 'relative' : 'absolute')};
  top: ${props => (props.isPinned ? '0' : 'auto')};
  right: ${props => (props.isPinned ? '0' : '1rem')};
  bottom: ${props => (props.isPinned ? '0' : '1rem')};

  display: flex;
  flex-direction: column;
`

export const Loading = styled.div`
  padding: 10px;
`

export const Error = Loading
