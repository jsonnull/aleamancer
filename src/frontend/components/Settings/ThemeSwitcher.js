// @flow
import React from 'react'
import styled from 'styled-components'
import Label from '../Label'
import Button from '../Button'
import type { ThemeName } from 'common/types'

const DarkButton = Button.extend`
  margin-left: 5px;
`

const ThemeButtons = styled.div`
  display: flex;
`

type Props = {
  currentTheme: ThemeName,
  changeTheme: ThemeName => any
}

const ThemeSwitcher = (props: Props) => {
  const isLightTheme = props.currentTheme == 'light'
  const isDarkTheme = props.currentTheme == 'dark'

  const useGreen = { green: true }
  const useOutline = { outline: true }

  const lightButton = isLightTheme ? useGreen : useOutline
  const darkButton = isDarkTheme ? useGreen : useOutline

  return (
    <div>
      <Label>Theme</Label>
      <ThemeButtons>
        <Button {...lightButton} onClick={() => props.changeTheme('light')}>
          Light
        </Button>
        <DarkButton {...darkButton} onClick={() => props.changeTheme('dark')}>
          Dark
        </DarkButton>
      </ThemeButtons>
    </div>
  )
}

export default ThemeSwitcher
