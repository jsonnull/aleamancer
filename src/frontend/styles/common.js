// @flow
export const fonts = {
  heading: 'Lora',
  body: `
    /* 1 */ -apple-system, BlinkMacSystemFont,
    /* 2 */ "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans",
    /* 3 */ "Helvetica Neue", sans-serif;
  `
}

export const fontSize = {
  small: '1.3rem',
  normal: '1.5rem',
  medium: '1.8rem',
  large: '2.5rem'
}

export const colors = {
  red: '#bf3542',
  yellow: '#ff974f',
  green: '#8c9b7a',
  blue: '#046380',
  lightBlue: '#057091',
  lightGray: '#e6e6e6'
}

export const timings = {
  easeOutQuad: 'cubic-bezier(0.25, 0.46, 0.45, 0.94)'
}

export const CONSTS = {
  headerHeight: '4.25rem'
}
