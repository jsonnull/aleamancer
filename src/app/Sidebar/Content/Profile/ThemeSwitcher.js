/* @flow */
import React from 'react'
import Label from '../Label'
import Button from 'app/components/Button'
import styles from './style.css'
import type { Theme } from 'types'

type ButtonProps = {
  selected: boolean,
  onClick: Function,
  children?: any
}
const ThemeButton = (props: ButtonProps) => {
  let variant = (props.selected) ? 'green' : 'outline'
  return <Button
      customClass={styles.themeButton}
      variant={variant}
      onClick={() => props.onClick()}>
    { props.children }
  </Button>
}

type Props = {
  currentTheme: Theme,
  changeTheme: (Theme) => any
}

const ThemeSwitcher = (props: Props) => {
  return <div>
    <Label>Theme</Label>
    <div className={ styles.themeButtons }>
      <ThemeButton
          selected={ props.currentTheme == 'light' }
          onClick={ () => props.changeTheme('light') }>
        Light
      </ThemeButton>
      <ThemeButton
          selected={ props.currentTheme == 'dark' }
          onClick={ () => props.changeTheme('dark') }>
        Dark
      </ThemeButton>
    </div>
  </div>
}

export default ThemeSwitcher
