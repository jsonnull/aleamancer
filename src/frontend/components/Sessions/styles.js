// @flow
import styled from 'styled-components'
import { fontSize, fonts } from 'frontend/styles/common'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex: 1;
`

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  background: ${props => props.theme.background};
  padding: 0 2rem;
`

export const Heading = styled.h1`
  font-size: ${fontSize.large};
  line-height: 1;
  font-family: ${fonts.heading};
  padding: 1rem 0;
  color: ${props => props.theme.color};
  margin: 0;
`
