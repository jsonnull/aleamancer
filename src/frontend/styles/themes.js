// @flow
import type { ThemeName } from 'common/types'
import { colors } from './common'

export type Theme = {
  name: ThemeName,
  background: string,
  backgroundSecondary: string,
  backgroundInverted: string,
  backgroundInvertedSecondary: string,
  color: string,
  colorSecondary: string,
  colorMedium: string,
  colorInverted: string,
  map: string,
  borderColor: string,
  modalOverlay: string,
  modalBackground: string
}

export const light: Theme = {
  name: 'light',
  background: '#f7f7f7',
  backgroundSecondary: '#ebe3d6',
  backgroundInverted: '#3c3c3c',
  backgroundInvertedSecondary: '#555',
  color: 'black',
  colorSecondary: '#3c3c3c',
  colorMedium: '#999',
  colorInverted: '#f7f7f7',
  map: '#bfbca5',
  borderColor: '#cdc5ba',
  modalBackground: '#f7f7f7',
  modalOverlay: 'rgba(0, 0, 0, 0.5)'
}

export const dark: Theme = {
  name: 'dark',
  background: '#3c3c3c',
  backgroundSecondary: '#2b2b2b',
  backgroundInverted: colors.blue,
  backgroundInvertedSecondary: colors.lightBlue,
  color: '#eee',
  colorSecondary: '#cdc5ba',
  colorMedium: '#999',
  colorInverted: '#eee',
  map: '#1a1a1a',
  borderColor: '#1a1a1a',
  modalBackground: '#2b2b2b',
  modalOverlay: 'rgba(0, 0, 0, 0.7)'
}
