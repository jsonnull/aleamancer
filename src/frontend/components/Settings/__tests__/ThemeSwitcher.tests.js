// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import ThemeSwitcher from '../ThemeSwitcher.js'
import { light } from 'frontend/styles/themes'

describe('Settings ThemeSwitcher component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={light}>
          <ThemeSwitcher currentTheme="light" changeTheme={() => {}} />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
