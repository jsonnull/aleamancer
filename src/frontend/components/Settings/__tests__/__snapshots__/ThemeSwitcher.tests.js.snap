// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings ThemeSwitcher component renders correctly 1`] = `
.c0 {
  font-size: 1.3rem;
  color: #3c3c3c;
  font-weight: bold;
  display: block;
}

.c2 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: #8c9b7a;
  box-shadow: inset 0 -20px 10px -10px #849273;
}

.c2:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

.c2:hover {
  background: #7f8d6f;
  box-shadow: inset 0 -20px 10px -10px #778468;
}

.c3 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: transparent;
  box-shadow: none;
  border: 1px solid black;
  color: black;
  margin-left: 5px;
}

.c3:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

.c3:hover {
  box-shadow: inset 0 -20px 10px -10px #d2d2d2;
  background: transparent;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <label
    className="c0"
  >
    Theme
  </label>
  <div
    className="c1"
  >
    <button
      className="c2"
      onClick={[Function]}
    >
      Light
    </button>
    <button
      className="c3"
      onClick={[Function]}
    >
      Dark
    </button>
  </div>
</div>
`;
